nucypher/config/storages.py

```diff
@@ -476,29 +475,24 @@ def remove(self, checksum_address: str, metadata: bool = True, certificate: bool
 
         if metadata is True:
             metadata_filepath = self.__generate_metadata_filepath(checksum_address=checksum_address)
-            os.remove(metadata_filepath)
+            metadata_filepath.unlink()
             self.log.debug("Deleted {} from the filesystem".format(checksum_address))
 
         if certificate is True:
             certificate_filepath = self.generate_certificate_filepath(checksum_address=checksum_address)
-            os.remove(certificate_filepath)
+            certificate_filepath.unlink()
             self.log.debug("Deleted {} from the filesystem".format(checksum_address))
 
         return
 
     def clear(self, metadata: bool = True, certificates: bool = True) -> None:
         """Forget all stored nodes and certificates"""
 
-        def __destroy_dir_contents(path) -> None:
-            try:
-                paths_to_remove = os.listdir(path)
-            except FileNotFoundError:
+        def __destroy_dir_contents(path: Path) -> None:
+            if not path.is_dir():
                 return
-            else:
-                for file in paths_to_remove:
-                    file_path = os.path.join(path, file)
-                    if os.path.isfile(file_path):
-                        os.unlink(file_path)
+            for dir_item in path.iterdir():
+                dir_item.unlink()
```

Comment:
To remain in line with the old code do we need to check for `is_file()` before unlinking? Unless the logic was already flawed and this fixes it...

Reply:
`known_nodes/certificates` and `known_nodes/metadata` are flat directories with files only so I thought having anything out of ordinary there should raise an error.

Reply:
Understood. I only pointed it out because of the original logic that was there. 

These are folders on the system that people can manipulate directly however they want eg. play around with directories/files (hopefully they don't 😅 ), and we probably don't want the node to necessarily crash/not function properly because the user created a folder there that doesn't directly affect the functioning of the node.

I would vote to keep the original logic that checks `isfile()` and perhaps log a warning if not a file, instead of raising an error.

Perhaps others have different opinions.

Reply:
I see your point, I think the original logic may even be extended to take more ownership of those directories:

```
        def __destroy_dir_contents(path: Path) -> None:
            if not path.is_dir():
                return
            for dir_item in path.iterdir():
                if dir_item.is_file():
                    dir_item.unlink()
                elif dir_item.is_dir():
                    __destroy_dir_contents(dir_item)
            path.rmdir()```

Reply:
In the context of what this PR is aiming to achieve, we shouldn't get too out of scope - perhaps something to look at in a follow-up PR - eg. original code did not delete the parent directory. I think simply checking for a file and unlinking could be done here (same as original code), and then you can look at a follow-up PR.